lib/pre-commit/utils/staged_files.rb

```diff
@@ -41,14 +75,16 @@ def binary?(file)
       end
 
       def filter_files(files)
-        files.reject do |file|
-          !File.exist?(file) ||
-          File.directory?(file) ||
-          (
-            size = File.size(file)
-            size > 1_000_000 || (size > 20 && binary?(file))
-          ) ||
-          repo_ignores.any? { |ignore| File.fnmatch?(ignore, file) }
+        first_pass = files
+          .reject { |file| repo_ignored?(file) }
+          .reject { |file| ignore_extension?(file) }
+          .reject { |file| File.directory?(file) }
+          .select { |file| File.exists?(file) }
```

Comment:
there is performance impact when using multiple `reject`/`select` instead of just one - especially on bigger lists, maybe we could set up on using a single `select`? (I think the reject was a bit more confusing as it's a negation logic)